### PR TITLE
Fix distribution rsample gradient tests on XPU/CUDA by using freeze_rng_state

### PR DESCRIPTION
**Pull request by gplutop7**
Fixes:
https://github.com/intel/torch-xpu-ops/issues/4020

`torch.get_rng_state()` only saves/restores CPU RNG state. On XPU/CUDA, `rsample()` draws from the device RNG, which is not restored, so `eps` and `rsample()` consume different random numbers and the gradient assertion fails.

`freeze_rng_state()` correctly freezes RNG state across all backends, fixing 6 tests: test_uniform, test_cauchy, test_halfcauchy, test_normal, test_exponential, test_laplace on XPU/CUDA.

This follows review feedback on PR #188079 to leverage existing PyTorch test utilities.

### CODE REVIEW

**Review by BBBela — APPROVED** [review]
Looks good to me, thanks! Although I have some considerations below.

Currently this change is not xpu-specific. That is a general PyTorch change that affects all the backends. I suppose we could follow the standard PyTorch flow, so simply create the xpu-independent issue in PyTorch and make this PR open for the Community review in the first place.

This PR potentially fixes those test cases also for CUDA and maybe also for MPS. Also, all those affected test cases are marked with `@expectedFailureMPS` and I guess the reason they are failing for XPU and MPS is the same and is fixed by this PR.

Overall it looks like a general PyTorch tests improvement 😉

**Comment by gplutop7** [comment]
Added one more freeze_rng_state() usage in the logisticnormal sampling test to stabilize RNG consumption across subcases

File: test/distributions/test_distributions.py [hunk 1]
```diff
@@ -2611,12 +2610,13 @@ def test_logisticnormal_sample(self):
             ref_dist.rvs = self._get_logistic_normal_ref_sampler(base_dist)
             mean_th = torch.tensor(mean)
             std_th = torch.tensor(np.sqrt(np.diag(cov)))
-            self._check_sampler_sampler(
-                LogisticNormal(mean_th, std_th),
-                ref_dist,
-                f"LogisticNormal(loc={mean_th}, scale={std_th})",
-                multivariate=True,
-            )
+            with freeze_rng_state():
```
Comment (etaf) [inline test/distributions/test_distributions.py]: Please keep the original behavior here.

**Review by etaf — CHANGES_REQUESTED** [review]

File: test/distributions/test_distributions.py (hunk 1 above)
Comment (gplutop7) [reply test/distributions/test_distributions.py]: I agree - we should keep the original behavior here. It freezes only PyTorch RNG, while NumPy/Python RNG (used in the reference path) still move forward, so it changes the test dynamics and can hide flakiness instead of testing true statistical behavior.

**Review by etaf — APPROVED** [review]

**Review by etaf — CHANGES_REQUESTED** [review]
Hi, @gplutop7  this PR is not a `fix`, could you update the PR description and make the motivation more clear? The change is good to me.

**Comment by gplutop7** [comment]
> Hi, @gplutop7 this PR is not a `fix`, could you update the PR description and make the motivation more clear? The change is good to me.

Thanks! Updated the PR title and description.

**Review by etaf — APPROVED** [review]
